easypoll/src/main/java/com/eis0/easypoll/poll/BinaryPoll.java

```diff
@@ -0,0 +1,300 @@
+    public static final String SELF_AUTHOR_NAME = "You";
+
+    private static long pollsCount = 0;
+    private static SharedPreferences mSharedPreferences;
+
+    private long number;
+    private String name;
+    private String question;
+    private Network author;
+    private Network users;
+    private int yesCount = 0;
+    private int noCount = 0;
+    private boolean closed = false;
+    private boolean incoming;
+
+    // ---------------------------- CONSTRUCTORS ---------------------------- //
+
+    /**
+     * Create a poll given all of its information. By default, every new poll is set as opened
+     * (it hasn't received all the answers yet).
+     *
+     * @param number The owner's local number of the poll.
+     * @param name The name of the poll.
+     * @param question The question asked to all users.
+     * @param author The one-peer {@link Network} representing the author of the poll.
+     * @param users The {@link Network} representing users subscribed to the poll.
+     * @throws IllegalArgumentException When one of the arguments is empty or different authors are given.
+     * @author Matteo Carnelos
+     */
+    public BinaryPoll(long number, @NonNull String name, @NonNull String question,
+                      @NonNull Network author, @NonNull Network users) {
+        if(name.isEmpty()) throw new IllegalArgumentException("Can't create poll with empty name.");
+        if(question.isEmpty()) throw new IllegalArgumentException("Can't create poll with empty question.");
+        if(author.size() > 1) throw new IllegalArgumentException("There can only be one poll author.");
+        if(users.isLocalNetwork()) throw new IllegalArgumentException("A poll must contain at least one user.");
+        this.number = number;
+        this.name = name;
+        this.question = question;
+        this.author = author;
+        this.users = users;
+        incoming = true;
+    }
+
+    /**
+     * Create a new poll from this device.<br>
+     * The poll number is calculated from the counter stored in the internal memory of the device.
+     * The author's {@link Network} is the local network, that is the one composed only with the
+     * local host, see {@link Network} for more details.
+     *
+     * @param name The name of the poll.
+     * @param question The question to ask all users.
+     * @param users Users to include in the poll.
+     * @throws IllegalArgumentException If one of the arguments is empty.
+     * @author Matteo Carnelos
+     */
+    public BinaryPoll(@NonNull String name, @NonNull String question, @NonNull Network users) {
+        this(pollsCount+1, name, question, NetworksPool.obtainLocalNetwork(), users);
+        incoming = false;
+        pollsCount++;
+        savePollsCountToInternal();
+    }
+
+    // ---------------------------- DATA STORING ---------------------------- //
+
+    /**
+     * Set the SharedPreferences reference to which save the {@link #pollsCount} value.
+     *
```

Comment:
Qui manca il @param.
E parlando più in generale per più o meno tutto il codice vedo che usi @NonNull ma solo occasionalmente, per coerenza ti consiglio di metterlo sempre e di sostituirlo con @Nullable se è il caso

Reply:
That's a good point, I missed **@param** here and some annotations along the code. I'll add them as soon as I can. I thought that annotations in overridden methods would be inherited automatically but after some searches I found out that it is not, they need to be repeated in every sub-method. Thanks!

Reply:
Perfect